src/Security/CheckGroups/CheckGroupDatabase.js

```diff
@@ -14,8 +14,15 @@ class CheckGroupDatabase extends CheckGroup {
   }
   setChecks() {
     const config = Config.get(Parse.applicationId);
+    let databaseUrl;
     const databaseAdapter = config.database.adapter;
-    const databaseUrl = databaseAdapter._uri;
+    if (databaseAdapter) {
+      // If database adapter is defined, use its URI
+      databaseUrl = databaseAdapter._uri;
+    } else if (config.databaseURI) {
+      // If database adapter is not defined, fallback to config.databaseURI
+      databaseUrl = config.databaseURI;
+    } 
```

Comment:
What is the behavior of Parse Server if both are defined? Is there a way to remove this logic and access the DB URI in another way? Because maybe no matter how the URI is defined, I would imagine it ends up in the same place?

Reply:
In the current implementation of both are defined then the adaptor uri will take precedence. And if the adapter uri is not defined then it will go for db uri.
Please let me know if the logic needs to change, And what will be the correct logic 
Thanks!

Reply:
> In the current implementation of both are defined then the adaptor uri will take precedence.

Is that so? It seems that Parse Server throws an error if both are defined:
https://github.com/parse-community/parse-server/blob/242002432fa305aa1c68b07679582164f560bda8/src/Controllers/index.js#L155-L161

I think - if possible - we should not replicate the logic here that is already defined in the code above. I suggest to pull the DB URI directly from the database controller. Let `getDatabaseController` handle the logic and only check the URI of the controller.

Reply:
https://github.com/parse-community/parse-server/blob/242002432fa305aa1c68b07679582164f560bda8/src/Controllers/index.js#L160-L167

Got it, so here's what I'm thinking:  In the above else if condition we can add one more condition to check if adaptor is assigned, if not we can assign the database URI to the adapter. This way, regardless of how the URI is set, it'll be considered. What do you think? Sounds like a plan to me!

Reply:
This sounds like duplicating the logic? Why not just get the DB controller and check the URI?

Reply:
Just a quick question for clarification: I took a look at the DB Controller class, but I couldn't spot any `DBURI `there. Could you help me understand how I should go about pulling the URI? Thanks for your help!

https://github.com/parse-community/parse-server/blob/242002432fa305aa1c68b07679582164f560bda8/src/Controllers/DatabaseController.js#L387-L403

Reply:
Did you try like it's already done: `databaseAdapter._uri;` where `_uri` is an internal var? But you would not get it from the config that is passed when initializing Parse Server but from the initialized Parse Server.